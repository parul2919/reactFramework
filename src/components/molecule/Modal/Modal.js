import React from "react";
import styled from "styled-components";
import styles from "./Modal.style";

const Modal = (props) => {
  const {
    className,
    isOpen,
    onClose,
    children,
    classes,
    overlaySide,
    isCloseOut,
  } = props;
  if (isOpen === false) return null;

  return (
    <div className={` col-sm-12 ${classes} ${className}`}>
      <div className={`modal-window  bigwidth open ${overlaySide}`}>
        <div
          className="modal-window-overlay"
          onClick={isCloseOut && onClose}
          role="button"
          aria-hidden="true"
        >
          .
        </div>
        <div className="modal-window-in comp-product-view">
          <div
            className="close"
            onClick={onClose}
            role="button"
            aria-hidden="true"
          >
            X
          </div>
          <div className="modal-window-in-content">{children}</div>
        </div>
      </div>
    </div>
  );
};

export default styled(Modal)`
  ${styles}
`;
