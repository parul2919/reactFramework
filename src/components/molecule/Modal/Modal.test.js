import React from "react";
import { render } from "@testing-library/react";
import Modal from "./Modal";

describe("<Component /> spec", () => {
  it("renders the component", () => {
    const view = render(<Modal />);
    expect(view).toMatchSnapshot();
  });
});
