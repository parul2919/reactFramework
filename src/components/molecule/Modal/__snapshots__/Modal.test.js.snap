// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" col-sm-12 undefined sc-bczRLJ ilfGBM"
      >
        <div
          class="modal-window  bigwidth open undefined"
        >
          <div
            aria-hidden="true"
            class="modal-window-overlay"
            role="button"
          >
            .
          </div>
          <div
            class="modal-window-in comp-product-view"
          >
            <div
              aria-hidden="true"
              class="close"
              role="button"
            >
              X
            </div>
            <div
              class="modal-window-in-content"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" col-sm-12 undefined sc-bczRLJ ilfGBM"
    >
      <div
        class="modal-window  bigwidth open undefined"
      >
        <div
          aria-hidden="true"
          class="modal-window-overlay"
          role="button"
        >
          .
        </div>
        <div
          class="modal-window-in comp-product-view"
        >
          <div
            aria-hidden="true"
            class="close"
            role="button"
          >
            X
          </div>
          <div
            class="modal-window-in-content"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
